### PR TITLE
87 subissue 32 create dashboard UI

### LINKED ISSUE #87: [Subissue 3.2] Create Dashboard UI
This is the new version of Subissue 3.2 that was created due to a shift in project direction.

### PR DESCRIPTION
**Pull request by Eggathin**
A fully functional DashboardUI that has yet to be connected to other UI Elements,
Allows the user to:
- Create a new curriculum
- Open an existing curriculum using its corresponding ID
- Displays a list of existing curriculums

Has yet to be connected to other UI elements, this will be corrected with a future PR

### CODE REVIEW

File: src/main/java/use_cases/display_curriculums/DisplayCurriculumsPresenter.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package use_cases.display_curriculums;
+
+import java.util.List;
+
+public class DisplayCurriculumsPresenter implements DisplayCurriculumsOutputBoundary {
+
+    private final List<DisplayCurriculumsInterface> viewObservers;
+
+    public DisplayCurriculumsPresenter(List<DisplayCurriculumsInterface> viewObservers) {
```
Comment (MMachadoUofT) [inline src/main/java/use_cases/display_curriculums/DisplayCurriculumsPresenter.java:9]: Good use of dependency inversion here! This should definitely help us adhere to clean architecture. I remember that circular issue we had, and this was such a good way to solve it while making sure our Controllers adhere to SRP. Good job!

**Review by Bhan-Aayush — APPROVED** [review]
The autograder was failing but now, I see you fixed the minor error. All the tests are passing, good job.

**Review by MMachadoUofT — APPROVED** [review]
Overall, this is excellent! I saw that you also connected this to our Main driver, meaning we've increased the functionality of our entire program with this. We're making awesome headway! We're only a few steps away from completing the program.

**Review by Amir1441 — APPROVED** [review]
Looks good, functionally should work. Also good work following Clean Architecture rules. Time to connect UI's.

File: src/main/java/dashboard_ui/DashboardUI.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    @Override
+    public void displayCurriculums(CurriculumsModel curriculumsModel) {
+        List<String> curriculums = new ArrayList<>();
+        Map<Integer, String> curriculumsMap = curriculumsModel.getCurriculums();
+        for (Integer id : curriculumsMap.keySet()) {
+            curriculums.add(String.format("%s (i.d. %s)", curriculumsMap.get(id), id));
+        }
+        Vector<String> curriculumsVector = new Vector<>(curriculums);
+        this.curriculumList.setListData(curriculumsVector);
+    }
```
Comment (Bhan-Aayush) [inline src/main/java/dashboard_ui/DashboardUI.java:92]: Good use of hashmap over here it stores the Integer and the string of the curriculum.

**Pull request merged**